lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+    "week_01_arm_1",
+    "week_02_arm_1",
+    "week_03_arm_1",
+    "week_04_arm_1",
+    "week_05_arm_1",
+    "week_06_arm_1",
+    "week_07_arm_1",
+    "week_08_arm_1",
+    "week_09_arm_1",
+    "week_10_arm_1",
+    "week_11_arm_1",
+    "week_11_arm_1",
+    "week_12_arm_1",
+    "week_13_arm_1",
+    "week_14_arm_1",
+    "week_15_arm_1",
+    "week_16_arm_1",
+    "week_17_arm_1",
+    "week_18_arm_1",
+    "week_19_arm_1",
+    "week_20_arm_1",
+    "week_21_arm_1",
+    "week_22_arm_1",
+    "week_23_arm_1",
+    "week_24_arm_1",
+    "week_25_arm_1",
+    "week_26_arm_1",
+]
+
+SWAB_AND_SEND_SITE = 'AIRSSwabNSend'
+
+STUDY_START_DATE = datetime(2021, 9, 9) # TODO: I dunno lol
+
+# TODO: are all these instruments required for enrollment?
+REQUIRED_ENROLLMENT_INSTRUMENTS = [
+    'screening',
+    'screening_call',
+    'informed_consent',
+    'enrollment',
+]
+
+
+def command_for_each_project(function):
+    """
+    A decorator to register one redcap-det subcommand per REDCap project, each
+    calling the same base *function*.
+
+    Used for side-effects only; the original *function* is unmodified.
+    """
+
+    for project in PROJECTS:
+        help_message = "Process REDCap DETs for Fred Hutch AIRS study"
+
+        redcap_det.command_for_project(
+            name = project.command_name,
+            redcap_url = REDCAP_URL,
+            project_id = project.id,
+            raw_coded_values = True,
+            revision = REVISION,
+            help = help_message,
+            include_incomplete = True)(function)
+
+    return function
+
+
+@command_for_each_project
+def redcap_det_fh_airs(*, db: DatabaseSession, cache: TTLCache, det: dict,
+    redcap_record_instances: List[REDCapRecord]) -> Optional[dict]:
+
+    if redcap_record_instances is None or len(redcap_record_instances) == 0:
+        LOG.warning(f"There are no record instances. Skipping record.")
+        return None
+
+    enrollments = [record for record in redcap_record_instances if record.event_name == ENROLLMENT_EVENT_NAME]
+
+    if not len(enrollments) == 1:
+        LOG.warning(f"There are {len(enrollments)} enrollment instances for record: {redcap_record_instances[0].get('subject_id')}. Skipping record.")
+        return None
+
+    enrollment = enrollments[0]
+
+    incomplete_enrollment_instruments = {
+                instrument
+                    for instrument
+                    in REQUIRED_ENROLLMENT_INSTRUMENTS
+                    if not is_complete(instrument, enrollment)
+            }
+
+    if incomplete_enrollment_instruments:
+        LOG.debug(f"The following required enrollment instruments «{incomplete_enrollment_instruments}» are not yet marked complete.")
+        return None
+
+    # If the participant's age < 18 do not continue
+    if (enrollment['scr_age'] == "" or int(enrollment['scr_age']) < 18):
+        LOG.debug("The participant age is blank or marked as under 18. Skipping record.")
+        return None
```

Comment:
I think we can take this part out because we'd want to leave this responsibility to ensure consent to the study itself. If there's redcap data, we should be able to assume there is proper consent.

Reply:
Noted and removed.